core/src/main/java/org/projectnessie/cel/extension/Gaurds.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2022 The Authors of CEL-Java
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.cel.extension;
+
+import java.util.function.BiFunction;
+import java.util.function.UnaryOperator;
+import org.projectnessie.cel.common.types.IntT;
+import org.projectnessie.cel.common.types.ListT;
+import org.projectnessie.cel.common.types.StringT;
+import org.projectnessie.cel.interpreter.functions.BinaryOp;
+import org.projectnessie.cel.interpreter.functions.FunctionOp;
+import org.projectnessie.cel.interpreter.functions.UnaryOp;
+
+/** function invocation guards for common call signatures within extension functions. */
+public class Gaurds {
+
+  private Gaurds() {}
+
+  public static BinaryOp callInStrIntOutStr(BiFunction<String, Integer, String> func) {
+    return (lhs, rhs) -> {
+      if (!(lhs instanceof StringT)) {
+        throw new RuntimeException();
```

Comment:
Throwing an exception isn't great here. Better return the appropriate error-value from one of the helpers in `org.projectnessie.cel.common.types.Err`

Reply:
Thanks. That has been addressed